.clang-format

```diff
@@ -0,0 +1,28 @@
+BasedOnStyle: LLVM
+IndentWidth: 8
+UseTab: ForIndentation
+BreakBeforeBraces: Attach
+AllowShortIfStatementsOnASingleLine: false
+IndentCaseLabels: false
+AlignAfterOpenBracket: Align
+AlignTrailingComments: true
+
+BreakBeforeBinaryOperators: NonAssignment
+
+MaxEmptyLinesToKeep: 1
+ColumnLimit: 140
```

Comment:
I would vote for 100 maximum to fit two files on the reasonable-sized screen next to each other.

Reply:
Editors can wrap, but they hardly ever unwrap. Me for example like to glance at the structure of the file on a casual screen - thus as long lines as meaningful. There was also an option to preserve some of the breaks in the source as well (so that if clauses would maintain the breaks on || && operators for example)

In the end it would be good if there was a tool to run and not have any comments on PR-s regarding whitespace and indentation and related friction :)

I do feel that 80 and nearby numbers are IMHO historical lowest common denominators that could be forgotten - people have ultrawide screens (even though I also work on a 15" laptop or even smaller net terminal _sometimes_ I would not define everything by these constraints)

Reply:
Who would have thought that formatting is an endless spring of discussion?

Let's start with something objective. The recommended formatting should match what's already used throughout most of the code. Currently, the line length is way below 140 lines for most of the code. I personally, think that 80 characters are great to force programmers to write shorter functions (that still fit into one screen).

Now something productive. The recommended formatting needs to be automatically forced onto pull request. There are some people using it in CI already, e.g. https://github.com/r-lib/usethis/pull/22/files.

Reply:
pre-commit hook?

Reply:
Seems like this got a bit stalled from last year. I would really like to see the formatting a bit more formalized in both documentation and automation. I just ran simple test through the opensc repository to find that there is really many lines even longer than the 140 characters (mostly legacy code, tables/lists or generated cmdline).
```
$ grep '.\{140\}' `find -name "*.[c|h]"` | wc -l
315
$ grep '.\{120\}' `find -name "*.[c|h]"` | wc -l
703
$ grep '.\{110\}' `find -name "*.[c|h]"` | wc -l
1184
$ grep '.\{100\}' `find -name "*.[c|h]"` | wc -l
2225
$ grep '.\{80\}' `find -name "*.[c|h]"` | wc -l
8219
```
I think we all agree that we do not want to go as low as 80. I would like to see 100, but I can settle up with 120 as it overflows only few characters when displayed in two columns on my 24" screen. :)